sql/000002.sql

```diff
@@ -0,0 +1,165 @@
+-- TODO evaluate perf re partial indexing instead of full unique
+alter table graphile_worker.jobs add column key text unique;
+
+alter table graphile_worker.jobs add locked_at timestamptz;
+alter table graphile_worker.jobs add locked_by text;
+
+-- update add_job behaviour to meet new requirements
+drop function graphile_worker.add_job;
+create function graphile_worker.add_job(
+  identifier text,
+  payload json = null,
+  queue_name text = null,
+  run_at timestamptz = null,
+  max_attempts int = null,
+  job_key text = null
```

Comment:
All optional arguments default to null because we need to be able to keep their existing values in the db, when doing an on conflict update - see coalesce calls below